docs/user-accounts.md

```diff
@@ -0,0 +1,94 @@
+
+In production, CCS staff log into RMI Admin using their Google account. They use
+RMI Admin to manage logins for external suppliers, which are stored in Auth0.
+Suppliers log into RMI Frontend, which talks to the RMI Admin API and to Auth0.
+
+          +--------------------+        +-------------------+
+          |     Google SSO     |        |       Auth0       |
+          | (RMI Admin logins) |        | (Supplier logins) |
+          +--------------------+        +-------------------+
+                      ^                       ^     ^
+                      |                       |     |
+                      |     +-----------------+     |
+                      |     |                       |
+                  +-----------+               +--------------+
+                  | RMI Admin |<--------------| RMI Frontend |
+                  +-----------+               +--------------+
+                        ^                             ^
+                        |                             |
+                    CCS staff                     Suppliers
+
+
+## RMI Admin
+
+In production, CCS staff log into RMI Admin using their Google SSO credentials.
+The set of users that are allowed access to the app is stored in the environment
+variable `ADMIN_EMAILS`.
+
+In development, the environment variable `ADMIN_EMAILS` is still used but no
+authentication is required. Just add your email address to `ADMIN_EMAILS` in
+`docker-compose.env` before starting the app, and you will be able to log in
+using that address.
+
+
+## RMI Frontend
+
+This is the app used by external suppliers to submit their returns. Logins to
+this service are authorized by the RMI Admin API, and supplier login details are
+stored in Auth0.
+
+Both RMI Admin and Frontend require credentials for connecting to Auth0's API.
+These credentials are stored in the `AUTH0_DOMAIN`, `AUTH0_CLIENT_ID` and
+`AUTH0_CLIENT_SECRET` environment variables (kept in `docker-compose.yml` in
+development). You'll need to get the right credentials for each app for the
+environment you're running in:
+
+- In development, we use the [dxw][1] tenant. It contains the following
+  applications:
+  - `Report MI Admin (Staging)`: corresponds to RMI Admin
+  - `Report MI (Staging)`: corresponds to RMI Frontend
+
+- In production, we use the [reportmi][2] tenant. It contains the following
+  applications:
+  - `Report MI Admin`
+  - `Report MI`
+
+To work with these services, you will need an admin login for Auth0, and access
+to at least the `dxw` tenant.
+
+So, to log into RMI Frontend in development:
+
+- Make sure your email is stored under [Users][3] in the `dxw` tenant
+- Make sure your details and Auth0 ID are stored in RMI Admin
+
+If your details are missing from either environment, you should be able to use
+the [Users section][4] of RMI Admin to add yourself. If your account already
+exists in dxw's Auth0 tenant, that should not be a problem. However, if you run
```

Comment:
I think it's the opposite. If your account _doesn't_ already exist in dxw's Auth0 tenant, then adding the user through the RMI Admin app will succeed, creating the user in the Admin DB and in Auth0. If the account _does_ exist, then it will fail to add it to Auth0, and it will display an error message and will not add the user to the Admin DB.

Reply:
I agree. If the user details are _only_ in the Auth0 tenant and we try to use the RMI Admin to create them then it will fail on uniqueness from within the Auth0 API.

Perhaps the merge of this fix https://github.com/dxw/DataSubmissionServiceAPI/pull/492 will make this moot as the statement will then become correct?

Reply:
> Perhaps the merge of this fix #492 will make this moot as the statement will then become correct?

I was just revisiting this PR and thinking exactly that! Happy to merge.